config/v1/types_dns.go

```diff
@@ -96,3 +103,31 @@ type DNSList struct {
 
 	Items []DNS `json:"items"`
 }
+
+// DNSPlatformSpec holds cloud-provider-specific configuration
+// for DNS administration.
+// +union
+type DNSPlatformSpec struct {
+	// type is the underlying infrastructure provider for the cluster.
+	// Allowed values are "AWS", "Azure", "BareMetal", "GCP", "Libvirt",
+	// "OpenStack", "VSphere", "oVirt", "KubeVirt", "EquinixMetal", "PowerVS",
+	// "AlibabaCloud", "Nutanix" and "None". Individual components may not support all platforms,
+	// and must handle unrecognized platforms as None if they do not support that platform.
+	//
+	// +unionDiscriminator
+	Type PlatformType `json:"type"`
```

Comment:
Please restrict this field to "" and the values that we currently support (I think just AWS).  The discriminated union was exactly the right choice, but not everything is valid.

Can this be different than the infrastructure?  If not, indicate that skewing infrastructure and this field is unsupported and be sure to code logic into the operator to indicate Degraded=true when they don't match.

Reply:
Question about this [here](https://github.com/openshift/api/pull/1460#discussion_r1203324260)